### PR TITLE
[v1] reduce graph capture time for piecewise cudagraph

### PR DESCRIPTION
**Pull request by youkaichao**
previously about 5 seconds for every shape, now it is less than 0.5 second.

### CODE REVIEW

File: vllm/compilation/backends.py [hunk 1]
```diff
@@ -503,17 +509,28 @@ def __call__(self, *args) -> Any:
...
-                    output = weak_ref_tensors(output)
+            with ExitStack() as stack:
+                if not self.is_first_graph:
+                    # during every model forward, we will capture
+                    # many pieces of cudagraphs (roughly one per layer).
+                    # running gc again and again across layers will
+                    # make the cudagraph capture very slow.
+                    # therefore, we only run gc for the first graph,
+                    # and disable gc for the rest of the graphs.
+                    stack.enter_context(patch("gc.collect", no_op))
```
Comment (WoosukKwon) [inline vllm/compilation/backends.py:520]: nit: 
```suggestion
                    stack.enter_context(patch("gc.collect", lambda: None))
```

**Review by WoosukKwon — APPROVED** [review]

File: vllm/compilation/backends.py (hunk 1 above)
Comment (youkaichao) [reply vllm/compilation/backends.py:520]: I explicitly add this. `no_op` is an existing function, while `lambda : None` will create a lambda function object everytime we call it.

Comment (youkaichao) [reply vllm/compilation/backends.py:520]: okay, never mind, this is cudagraph capture time rather than runtime, it does not make any difference here.

Comment (youkaichao) [reply vllm/compilation/backends.py]: fixed in [708da06](https://github.com/vllm-project/vllm/pull/10059/commits/708da06650c8479024e7a2d2213ab15afad7ebce)

**Pull request merged**